### ISSUE TITLE
Support declarative configuration of OCI hooks in YAML

### ISSUE THREAD

**Issue by deliciouslytyped** [containers/podman-compose#794]
It would be nice if there was a way to specify OCI hooks / lifecycle hooks in the podman compose yaml. This would allow extending the container lifecycle.

I've taken some quick notes in https://github.com/hercules-ci/arion/issues/218 , mainly on OCI hooks, discussion on those points should probably be imported here.
https://github.com/containers/podman-compose/issues/278 is related.

**Comment by MoralCode** [comment]
+1 This seems like it is very useful for enabling applications to run in openshift (where random user IDs are assigned to containers and they aren't usually run as root). Especially if a permissions setting command needs to run as root to ensure, for example, a logs directory is writable

**Comment by Anachron** [comment]
Yeah I would love to use hooks inside my yaml to run a `modprobe` before container start and a `rmmod` on container shutdown.
